### PR TITLE
Add support for openssl3

### PR DESCRIPTION
**Pull request by adfoster-r7**
Relates to https://github.com/rapid7/metasploit-framework/issues/16818

Updating this repository to support OpenSSL 3

## Verification

- Code review
- Verify CI passes

### CODE REVIEW

File: app/models/metasploit/credential/ntlm_hash.rb [hunk 1]
```diff
@@ -1,5 +1,22 @@
 require 'net/ntlm'
 
+# TODO: Revert once available in rubyntlm
+# https://github.com/WinRb/rubyntlm/pull/51
```
Comment (adfoster-r7) [inline app/models/metasploit/credential/ntlm_hash.rb:4]: Temporarily applying the patch from https://github.com/WinRb/rubyntlm/pull/51 until it's available as a release

Comment (adfoster-r7) [reply app/models/metasploit/credential/ntlm_hash.rb:4]: cc @jmartin-r7 Are we happy with this patch being here? I envision rubyntlm will take a long time to be fixed properly upstream

The unit tests caught this edgecase for us

https://github.com/rapid7/metasploit-credential/blob/ac481e8b05b8bcd878d8c88f02676be3eed4e1bf/spec/models/metasploit/credential/ntlm_hash_spec.rb#L354

File: spec/factories/metasploit/credential/ssh_keys.rb [hunk 2]
```diff
@@ -4,7 +4,7 @@
     transient do
       key_type { generate :metasploit_credential_ssh_key_key_type }
       # key size tuned for speed.  DO NOT use for production, it is below current recommended key size of 2048
-      key_size { 512 }
+      key_size { 1024 }
```
Comment (adfoster-r7) [inline spec/factories/metasploit/credential/ssh_keys.rb:7]: OpenSSL 3 doesn't like the small key size:

```
irb(main):006:0> OpenSSL::PKey::DSA.new(512)
/usr/lib/ruby/3.0.0/openssl/pkey.rb:173:in `generate_key': EVP_PKEY_keygen: bad ffc parameters (OpenSSL::PKey::PKeyError)
	from /usr/lib/ruby/3.0.0/openssl/pkey.rb:173:in `generate'
	from /usr/lib/ruby/3.0.0/openssl/pkey.rb:180:in `new'
	from (irb):6:in `<main>'
	from /var/lib/gems/3.0.0/gems/irb-1.3.6/exe/irb:11:in `<top (required)>'
	from /usr/local/bin/irb:25:in `load'
	from /usr/local/bin/irb:25:in `<main>'
```

Works with 1024 on OpenSSL 3:

```
irb(main):007:0> OpenSSL::PKey::DSA.new(1024)
=> #<OpenSSL::PKey::DSA:0x000055f23a9f6258 oid=DSA>
```

File: app/models/metasploit/credential/ntlm_hash.rb (hunk 1 above)
Comment (jmartin-r7) [reply app/models/metasploit/credential/ntlm_hash.rb:4]: This seems reasonable.

**Review by jmartin-r7 — APPROVED** [review]

**Pull request merged**